lib/cli/drive-scanner.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const drivelist = require('drivelist');
+const EventEmitter = require('events').EventEmitter;
+const _ = require('lodash');
+const Rx = require('rx');
+
+const interval = new WeakMap();
+const prevPollDrives = new WeakMap();
+const drivesStream = new WeakMap();
+
+const createDrivesStream = (scanInterval) => {
+  return Rx.Observable.timer(0, scanInterval)
+    .flatMap(() => {
+      return Rx.Observable.fromNodeCallback(drivelist.list)();
+    })
+    .pausable(new Rx.Subject());
+};
+
+const emitChangeEvent = function(drives) {
+  const prevDrives = prevPollDrives.get(this);
+  if (prevDrives !== null && _.isEqual(drives, prevPollDrives)) {
+    return;
+  }
+  this.emit('change', drives, prevDrives);
```

Comment:
I can move it in the class. The reason I extracted it is because I thought it would make the constructor more readable. Once again, I created it having a private method in mind and it made sense to me to be here as one of the `DriveScanner`'s private methods, and it looked like it won't cause damage because this whole module exists just to accomodate the `DriveScanner`.
On the other hand, I admit that I don't know if that's the way to do things in JS or not, maybe it's an anti-pattern. I think I need to read some more on the subject of private instance methods and variables in JS best practices and patterns.